core-bundle/src/Job/Job.php

```diff
@@ -73,13 +73,18 @@ public function getUuid(): string
         return $this->uuid;
     }
 
-    public function withProgressFromAmounts(int $total, int $amount): self
+    public function withProgressFromAmounts(int $amount, int|null $total = null): self
     {
         // Prevent division by 0
         if (0 === $total) {
             return $this;
         }
 
+        // Sometimes the total is not known, let's just estimate for our progress
+        if (null === $total) {
+            $total = $amount * random_int(110, 200) / 100; // do not use a hardcoded number as otherwise progress would always show 50%
```

Comment:
So when `$amount` is `5` in the first call and `random_int()` returns `120`, `$total` will be `6` and `$progress` will be `83.33`.

Then in the next call, `$amount` is `6` but `random_int()` returns `180`, so `$total` will be `10.9` and `$progress` will be only `55.05`.

This would make the progress bar go backwards, wouldn't it?